packages/core/saved-objects/core-saved-objects-utils-server/src/saved_objects_test_utils.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { isEqual } from 'lodash';
+
+/**
+ * Determines if a given Set is equal to another given Set. Set types must be the same, and comparable.
+ *
+ * @param setA the first Set to compare
+ * @param setB the second Set to compare
+ * @returns {boolean} True if Set A is equal to Set B
+ */
+export function setsAreEqual<T>(setA: Set<T>, setB: Set<T>) {
+  // console.log(`*** SET A: ${Array.from(setA)}`);
+  // console.log(`*** SET B: ${Array.from(setB)}`);
+  return isEqual(Array.from(setA).sort(), Array.from(setB).sort());
+}
+
+/**
+ * Determines if a given type map is equal to another given type map.
+ *
+ * @param mapA the first type map to compare
+ * @param mapB the second type map to compare
+ * @returns {boolean} True if type map A is equal to type map B
+ */
+export function typeMapsAreEqual(mapA: Map<string, Set<string>>, mapB: Map<string, Set<string>>) {
+  return (
+    mapA.size === mapB.size &&
+    isEqual(Array.from(mapA!.keys()).sort(), Array.from(mapB!.keys()).sort()) &&
```

Comment:
NIT: Just use `setsAreEqual` instead of manually reproducing it

Reply:
keys() isn't a set (it's an IterableIterator) so it would need to be converted to call setsAreEqual, but I realized I can simplify this like so...
```
 return (
    mapA.size === mapB.size &&
    Array.from(mapA.keys()).every((key) => mapB.has(key) && setsAreEqual(mapA.get(key)!, mapB.get(key)!))
  );
```
This does still use an array conversion, but there has to be a conversion of some kind to implement either a forEach or every. The only other alternative I see, with no conversions, is this:
```
  let isEqual = mapA.size === mapB.size;
  let iterator = mapA.keys().next();
  while (isEqual && !iterator.done) {
    isEqual &&= mapB.has(iterator.value) && setsAreEqual(mapA.get(iterator.value)!, mapB.get(iterator.value)!);
    iterator = mapA.keys().next();
  }
  return isEqual;
```
But that's a bit less readable. What do you think?